dvc/remote/gdrive.py

```diff
@@ -272,18 +315,42 @@ def _list_params(self):
         if self._bucket != "root" and self._bucket != "appDataFolder":
             drive_id = self._gdrive_shared_drive_id(self._bucket)
             if drive_id:
+                logger.debug(
+                    "GDrive remote '{}' is using shared drive id '{}'.".format(
+                        self.path_info, drive_id
+                    )
+                )
                 params["driveId"] = drive_id
                 params["corpora"] = "drive"
         return params
 
     @_gdrive_retry
     def _gdrive_shared_drive_id(self, item_id):
+        from pydrive2.files import ApiRequestError
+
         param = {"id": item_id}
         # it does not create a file on the remote
         item = self._drive.CreateFile(param)
         # ID of the shared drive the item resides in.
         # Only populated for items in shared drives.
-        item.FetchMetadata("driveId")
+        try:
+            item.FetchMetadata("driveId")
+        except ApiRequestError as exc:
+            error_code = exc.error.get("code", 0)
+            if error_code == 404:
+                raise DvcException(
+                    "'{}' for '{}':\n\n"
+                    "1. Check that directory exists and you can access it.\n"
+                    "2. Make sure that credentials in '{}'\n"
+                    "   are the right ones to use with this remote storage, "
+                    "e.g. use 'gdrive_user_credentials_file' config\n"
+                    "   option if you use multiple GDrive remotes with "
+                    "different email accounts.\n\nDetails".format(
```

Comment:
```suggestion
                    "1. Check that the directory exists and you can access it.\n"
                    "2. Make sure that credentials in '{}'\n"
                    "   are correct for this remote e.g. "
                    "use the `gdrive_user_credentials_file` config\n"
                    "   option if you use multiple GDrive remotes with "
                    "different email accounts.\n\nDetails".format(
```
> Needs rewrap

But do we really need the last part `e.g. use the ...` ? Feels a little long.

Reply:
"Check that **the** directory"
"are the right ones to use with" -> "are correct for" seems more direct and simple? 🙂 
Also, I see "remote storage" as a concept, when specifying "a remote" or "a DVC remote" I try not to use the word "storage".
Back quotes

Reply:
@jorgeorpinel I already applied almost all your suggestions before resolving :) please, refresh the page?

Reply:
Oh. Weird. It didn't refresh earlier, weird. OK, thanks! The first sentence is still missing a word though: "Check **that** the directory" 🙂

Reply:
it can work w/o that (I googled both options) - shorter version allows better code (not need to break the line one more time)

Reply:
"Check the directory" works by itself, but it's a slightly different meaning, to continue the phrase I believe "that" is needed (i.e. check **that** the dir something") if we're being strict. Otherwise "Confirm the dir exists" would sound better to me.

Not a big deal obviously but... user before developer 😬

Reply:
Changed to confirm!